tests/ui/lang/core/array/oob_0_array.rs

```diff
@@ -0,0 +1,11 @@
+#![allow(unconditional_panic)]
+
+#![cfg_attr(target_arch = "spirv", no_std)]
+use spirv_std::spirv;
+
+#[spirv(compute(threads(1, 1, 1)))]
+pub fn compute() {
+    let mut array = [0; 0];
+    // for some reason this now compiles?
```

Comment:
We should look into this case, no?

Reply:
yea, I want to see if it compiles to include a panic, but I'm unsure of how to run the rest to see what happens

Is it run automatically by compiletest

Reply:
No, sadly compile test just checks it compiles. We don't have any Automated runtime tests right now sadly. I usually just drop it in an example when I am doing a quick test.

Reply:
I've checked the output spirv assembly for that example (by adding the compile flags), and it seems that the assignment is getting optimized out somehow. I checked all the spirv builder methods, and none of them seem to be hit. I've found that any reads from the array hit gep and fail, but stores do not. I feel like ideally that should give a runtime panic, but I'm not entirely sure where to inject it.

While investigating this, I also noticed that zst arrays as input parameters do not work.`&mut [T; 0]` also gives the old cryptic error message. I'm not sure if I should fix that in this PR, or another

Reply:
Is the store behavior changed by this PR? If it is unchanged I'd say just leave it and file a followup task.

Feel free to fix the input message in this PR if you want.

Reply:
Not sure if this is answered or ready to review, ping if so!

Reply:
Oops I forgot to reply because of the holidays, I checked but the behavior isn't changed, this previously also compiled on main